contracts/test/TotalSupply.js

```diff
@@ -0,0 +1,27 @@
+"strict mode"
+
+
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+
+const SUDT_ID = 1;
+const SUDT_NAME = 'ckETH';
+const SUDT_SYMBOL = 'ckETH';
+const SUDT_DECIMALS = 18;
+const SUDT_TOTAL_SUPPLY = 9999999999;
+
+describe("TotalSupply", function () {
+  it("Deoloy erc20 proxy -> get total supply", async () => {
+    const erc20Fact = await ethers.getContractFactory("contracts/SudtERC20Proxy_UserDefinedDecimals.sol:ERC20");
+    const erc20 = await erc20Fact.deploy(SUDT_NAME, SUDT_SYMBOL, SUDT_TOTAL_SUPPLY, SUDT_ID, SUDT_DECIMALS);
+    console.log(`  ERC20 on address: ${erc20.address}`);
+    const [sender] = await ethers.getSigners();
+    const totalSupply = await erc20.callStatic.totalSupply();
+    expect(totalSupply, SUDT_TOTAL_SUPPLY);
```

Comment:
It would be better to run deposits and withdrawals, and check if `totalSupply` changes accordingly.

see also: [test_deposit_and_withdrawal in Godwoken](https://github.com/nervosnetwork/godwoken/pull/560/files#diff-ba9cba5e76a9b5be5068d033b5073145c513af3dd6048306cd26aaaafd2f9ed2R112)